io.asgardeo.tomcat.oidc.agent/src/main/java/io/asgardeo/tomcat/oidc/agent/OIDCAgentFilter.java

```diff
@@ -136,27 +141,32 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         }
     }
 
-    private String resolveTargetPage(RequestContext requestContext) {
+    private String resolveTargetPage(HttpServletRequest request, RequestContext requestContext) {
 
-        String targetPage = "home.jsp";
+        String targetPage = "";
         if (StringUtils.isNotBlank(oidcAgentConfig.getHomePage())) {
             targetPage = oidcAgentConfig.getHomePage();
-        } else if (requestContext != null) {
-            if (requestContext.getParameter(SSOAgentConstants.REDIRECT_URI_KEY) != null) {
-                targetPage = requestContext.getParameter(SSOAgentConstants.REDIRECT_URI_KEY).toString();
-            }
+        } else if (requestContext != null && StringUtils.isNotBlank((CharSequence) requestContext.getParameter(
+                SSOAgentConstants.REDIRECT_URI_KEY))) {
+            targetPage = requestContext.getParameter(SSOAgentConstants.REDIRECT_URI_KEY).toString();
+        } else if (StringUtils.isNotBlank(oidcAgentConfig.getIndexPage())) {
+            targetPage = oidcAgentConfig.getIndexPage();
+        }  else {
+            String requestUrl = request.getRequestURL().toString();
+            targetPage = requestUrl.substring(0, requestUrl.length() - request.getServletPath().length());
```

Comment:
Just a suggestion, can we think of replacing this if-else with if and return blocks to make it readable :)

Reply:
Addressed